### PR TITLE
Resolving ROI issue & resolving bug in fused_crop_decoder 

### PR DESCRIPTION
**Pull request by swetha097**
This PR resolves

1. Resolves the ROI issue
2. Bug in the fused crop decoder when the x & x_diff coordinates are not same

### CODE REVIEW

File: rocAL/rocAL/source/decoders/image/fused_crop_decoder.cpp [hunk 1]
```diff
@@ -138,40 +138,39 @@ Decoder::Status FusedCropTJDecoder::decode(unsigned char *input_buffer, size_t i
             y1 =  (original_image_height - crop_height) / 2;
         }
     }
-    
-   // std::cout<<"Fused Crop Decoder <x,y, w, h>: " << x1 << " " << y1 << " " << crop_width << " " << crop_height << std::endl;
+    // Handles the case where generated crop width / crop height can be greater than the max decoded width & max decoded height set by the user.
+    if(crop_width > max_decoded_width)
```
Comment (rrawther) [inline rocAL/rocAL/source/decoders/image/fused_crop_decoder.cpp:142]: please use std::min(crop_width, max_decoded_width) instead and similar for crop_height

File: rocAL/rocAL/source/decoders/image/fused_crop_decoder.cpp [hunk 2]
```diff
@@ -138,40 +138,39 @@ Decoder::Status FusedCropTJDecoder::decode(unsigned char *input_buffer, size_t i
...
+                             input_size,
+                             output_buffer,
+                             max_decoded_width,
+                             max_decoded_width * planes,
+                             max_decoded_height,
+                             tjpf,
+                             TJFLAG_FASTDCT, &x1_diff, &crop_width_diff,
+		             x1, y1, crop_width, crop_height) != 0)
 
     {
```
Comment (rrawther) [inline rocAL/rocAL/source/decoders/image/fused_crop_decoder.cpp:158]: please move the { to previous line

File: rocAL/rocAL/source/decoders/image/fused_crop_decoder.cpp [hunk 3]
```diff
@@ -138,40 +138,39 @@ Decoder::Status FusedCropTJDecoder::decode(unsigned char *input_buffer, size_t i
...
+        unsigned int xoffs = (x1 - x1_diff) * planes;   // in case x1 gets adjusted by tjpeg decoder
+        src_ptr_temp = output_buffer;
         dst_ptr_temp = output_buffer;
         for (unsigned int i = 0; i < crop_height; i++)
         {
             memcpy(dst_ptr_temp, src_ptr_temp + xoffs, elements_in_crop_row * sizeof(unsigned char));
-            //memset(dst_ptr_temp + elements_in_crop_row, 0, remainingElements * sizeof(unsigned char));
-            src_ptr_temp +=  elements_in_row;
-            dst_ptr_temp +=  elements_in_row;
+            src_ptr_temp +=  xoffs + elements_in_crop_row;
```
Comment (rrawther) [inline rocAL/rocAL/source/decoders/image/fused_crop_decoder.cpp:172]: Not sure if this is the right math. The stride is elements_in_row which may not be same as (elements_in_row)

**Review by rrawther — COMMENTED** [review]
@swetha097 please address review comments

**Comment by rrawther** [comment]
@swetha097 : Is this PR still required. You mentioned it is fixed with the correct version of TJpeg decoder?

**Comment by kiritigowda** [comment]
@LakshmiKumar23 @swetha097 any updates on this PR?

**Comment by swetha097** [comment]
Closing the PR as it is fixed with the correct version of turbo jpeg

**Pull request closed without merging**